### PR TITLE
Fixed panic in `prepare_input_samples`.

### LINKED ISSUE #1192: Audio mixer's `prepare_inputs` sometimes crashes
```
2025-07-31T09:03:28.887068Z  INFO Audio decoder thread: compositor_pipeline::pipeline::resampler::dynamic_resampler: Initializing stereo audio resampler input_sample_rate=44100 output_sample_rate=48000 input_id="input_1" decoder="FDK AAC decoder"

thread '<unnamed>' panicked at compositor_pipeline/src/audio_mixer/prepare_inputs.rs:143:54:
slice index starts at 1019 but ends at 960
```

### PR DESCRIPTION
**Pull request by JBRS307**
Closes #1192

### CODE REVIEW

File: compositor_pipeline/src/audio_mixer/prepare_inputs/single_frame_tests.rs [hunk 1]
```diff
@@ -364,4 +364,35 @@ fn test_prepare_inputs() {
             (5.0, 5.0)
         ]
     );
+
+    // Severly missaligned timestamps (to the left)
```
Comment (wkozyra95) [inline compositor_pipeline/src/audio_mixer/prepare_inputs/single_frame_tests.rs]: misaligned is a very generic statement here,  for once it's subjective it's also not really true, it does not matter for this bug how much batch is outside of the exepected ranage

if you want to describe what this test case is testing, then it should be something like this:
```
The first batch ends before the result batch starts by at least one sample duration
```

Also, if you really want to show the corner case, you can use 
```
let first_batch_start = start - (5 * sample_duration) - small_error;
```
instead of
```
let first_batch_start = start - batch_duration;
```

btw I'm assuming here that this was the corner case, don't know that for sure

File: compositor_pipeline/src/audio_mixer/prepare_inputs.rs [hunk 2]
```diff
@@ -129,7 +129,7 @@ fn frame_input_samples(
                     samples_to_remove_from_start
```
Comment (wkozyra95) [inline compositor_pipeline/src/audio_mixer/prepare_inputs.rs:130]: If first batch is dropped entirely, then this warning can give false positve because we might need to drop some samples from the second batch too
